internal/pkg/crypto/randomness.go

```diff
@@ -0,0 +1,66 @@
+package crypto
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/crypto"
+	"github.com/minio/blake2b-simd"
+	"github.com/pkg/errors"
+)
+
+// RandomnessSource provides randomness to actors.
+type RandomnessSource interface {
+	Randomness(tag crypto.DomainSeparationTag, epoch abi.ChainEpoch, entropy []byte) (abi.Randomness, error)
+}
+
+type RandomSeed []byte
+
+type ChainSampler interface {
+	Sample(epoch abi.ChainEpoch) (RandomSeed, error)
+}
+
+// A randomness source that seeds computations with a sample drawn from a chain epoch.
+type ChainRandomnessSource struct {
+	Sampler ChainSampler
+}
+
+func (c *ChainRandomnessSource) Randomness(tag crypto.DomainSeparationTag, epoch abi.ChainEpoch, entropy []byte) (abi.Randomness, error) {
+	seed, err := c.Sampler.Sample(epoch)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to sample chain for randomness")
+	}
+	return blendEntropy(tag, seed, entropy)
+}
+
+// A randomness source for use when computing genesis state (the state that the genesis block points to as parent state).
+// There is no chain to sample a seed from.
+type GenesisRandomnessSource struct{}
+
+func (g *GenesisRandomnessSource) Randomness(tag crypto.DomainSeparationTag, epoch abi.ChainEpoch, entropy []byte) (abi.Randomness, error) {
+	if epoch > 0 {
+		return nil, fmt.Errorf("invalid use of genesis randomness source for epoch %d", epoch)
+	}
+	seed := []byte{}
```

Comment:
again, I think this is wrong.

Reply:
Insofar as I understand it, we will want to have a specific pseudo-random value for use so the initial set of miners can preSeal their sectors for genesis. I don't see a reason why it couldn't be the one you define above (though there may be some reason to keep it secret until a chosen moment?). Just need to make sure it's the same as w lotus...